doc/doxygen/Makefile

```diff
@@ -7,39 +7,39 @@ export STRIP_FROM_INC_PATH_LIST=$(shell \
 
 # use lessc (http://lesscss.org/#using-less) for compiling CSS, alternatively
 # fall back to lesscpy (https://github.com/lesscpy/lesscpy)
-ifeq (,$(LESSC))
-  ifneq (,$(shell command -v lessc 2>/dev/null))
-    LESSC=lessc
-  else ifneq (,$(shell command -v lesscpy 2>/dev/null))
-    LESSC=lesscpy
-  endif
+ifneq (,$(shell command -v lessc 2>/dev/null))
+  LESSC_DEFAULT=lessc
+else ifneq (,$(shell command -v lesscpy 2>/dev/null))
+  LESSC_DEFAULT=lesscpy
 endif
 
+LESSC ?= $(LESSC_DEFAULT)
+
 .PHONY: doc
 doc: html
 
 # by marking html as phony we force make to re-run Doxygen even if the directory exists.
 .PHONY: html
-html: src/css/riot.css src/changelog.md
+html: src/changelog.md
 	( cat riot.doxyfile ; echo "GENERATE_HTML = yes" ) | doxygen -
 
 .PHONY: man
 man: src/changelog.md
 	( cat riot.doxyfile ; echo "GENERATE_MAN = yes" ) | doxygen -
 
-ifneq (,$(LESSC))
-src/css/riot.css: src/css/riot.less src/css/variables.less
-	@$(LESSC) $< $@
+.PHONY: riot-css
+riot-css: src/css/variables.less
+	@$(LESSC) src/css/riot.less src/css/riot.css
```

Comment:
This is somewhat ugly now with input files which are no longer dependencies of the target. How about only adding this to the original recipe specification?
```make
.PHONY: riot-css
riot-css: src/css/riot.css
```

The removal of src/css/riot.css from the prerequisites of the `html` target should give the desired results anyway.

Reply:
you may want to list src/css/riot.css as .PHONY to get unconditional rebuild if that was desired.

Reply:
`src/css/variables.less` should be prerequisite so that it gets updated from `src/config.json` each time `riot-css` is run.

Making `src/css/riot.css` a dependency of `riot-css` is the same as nothing, since the PHONY rule will be run anyways. The only difference is what happens when `src/css/riot.css` is missing (in one case `lessc` fails and in the other `make` fails before calling `lessc`).

Reply:
why is riot.less not a prerequisite anymore?

Reply:
Because

1. riot-css is a PHONY target, so it will always get redone if the user types `make riot-css`
2. `riot.less` already exists and there is no way to "rebuild it".

Point (2) is the difference between `riot.less` and `variables.less`. The latter is automatically generated from a json, so make has to know that riot-css depends on it so that if the json changes, `variables.less` before remaking `riot.css`.

If I were to add `riot.less` to the prerequisites of `riot-css` it would not change anything because there is no rule to remake `riot.less`, make always considers it to be up to date.

I can do it anyways if it improves clarity.

Reply:
>I can do it anyways if it improves clarity.

I would prefer it. There is no reason to remove correct dependencies even though riot.css is always rebuilt.

@cladmi what was the conclusion regarding .PHONY: deps vs recipe: FORCE? should riot.css depend on FORCE instead of being .PHONY?

Reply:
If the name of the target was not `riot-css` but rather `src/css/riot.css` then it should be FORCE and not PHONY since it corresponds to a real file.

Here the target name is _intentionally_ different from the file so that make never attempts to rebuild the file by itself. Then it should be PHONY.

Off topic: the most important conclusions of PHONY vs FORCE are that:
- For file-targets FORCE should be used instead of PHONY.
- If some target is being FORCEd, there is a high probability that the makefile has a design issue and that someone is sweeping problems under the rug.